src/futils.c

```diff
@@ -26,27 +26,25 @@ int git_futils_mkpath2file(const char *file_path, const mode_t mode)
 
 int git_futils_mktmp(git_str *path_out, const char *filename, mode_t mode)
 {
-	int fd;
-	mode_t mask;
-
-	p_umask(mask = p_umask(0));
+	const int open_flags = O_RDWR | O_CREAT | O_EXCL | O_BINARY | O_CLOEXEC;
+	int fd = -1;
+	unsigned int tries = TMP_MAX;
```

Comment:
TMP_MAX isn't really related to mktemp, but there also isn't any suggested alternative